### PR TITLE
Jiminy Support for at86rfr2 transceiver

### PR DESCRIPTION
**Pull request by Josar**
This PR adds support for the at86rfr2 transceiver into the at86rf2xx driver.

This is split out from https://github.com/RIOT-OS/RIOT/pull/8742

### CODE REVIEW

**Comment by Josar** [comment]
@kYc0o maybe this could be a review where you could help out as you could test it on the RCB256RFR2-xpro. 😁

**Comment by PeterKietzmann** [comment]
I don't want / can't hop into the review process so late. Just wanted to inform that I can test on the Jiminy hardware once we are at that point.

**Comment by kYc0o** [comment]
Yes, but first I'd like to get #9130 in. Afterwards I want to add support for that board, then I can test this.

If the guys at HAW who have the jiminy board can test it before it's also OK.

**Comment by Josar** [comment]
@gebart could you find some time for this?

File: drivers/at86rf2xx/at86rf2xx_netdev.c [hunk 1]
```diff
@@ -130,59 +148,115 @@ static int _send(netdev_t *netdev, const iolist_t *iolist)
...
-    /* just return length when buf == NULL */
+    /* just return length when buf == NULL, Upper layer calls _recv
+     * with no buffer to get expected size for allocating buffer
+     *
+     * Attetion:fb_stop does remove the frame buffer protection !
+     * */
     if (buf == NULL) {
+        /* TODO Is this not a problem? Frame Buffer release */
         at86rf2xx_fb_stop(dev);
+        DEBUG("Packet lenght: %d\n", pkt_len);
```
Comment (kYc0o) [inline drivers/at86rf2xx/at86rf2xx_netdev.c]: @miri64 or @thomaseichinger, do you have an opinion on this?

File: drivers/at86rf2xx/at86rf2xx_netdev.c [hunk 2]
```diff
@@ -130,59 +148,115 @@ static int _send(netdev_t *netdev, const iolist_t *iolist)
...
         return pkt_len;
     }
     /* not enough space in buf */
     if (pkt_len > len) {
         at86rf2xx_fb_stop(dev);
+#ifdef MODULE_AT86RFR2
+        /* If Upper layer does not supply a big enough buffer data loss is
+         * unavoidable clear frame buffer protection
+         * */
+        *AT86RF2XX_REG__TRX_CTRL_2 &= ~(1 << RX_SAFE_MODE);
```
Comment (kYc0o) [inline drivers/at86rf2xx/at86rf2xx_netdev.c]: You disable the FB protection here, but you also enable it for any device. I think this may cause conflicts...

File: drivers/at86rf2xx/at86rf2xx_netdev.c [hunk 3]
```diff
@@ -130,59 +148,115 @@ static int _send(netdev_t *netdev, const iolist_t *iolist)
...
+#endif
         return -ENOBUFS;
     }
     #ifdef MODULE_NETSTATS_L2
-        netdev->stats.rx_count++;
-        netdev->stats.rx_bytes += pkt_len;
+    netdev->stats.rx_count++;
+    netdev->stats.rx_bytes += pkt_len;
```
Comment (kYc0o) [inline drivers/at86rf2xx/at86rf2xx_netdev.c]: I'd say you need also to remove the indentation of the `#ifdef` statement.

File: drivers/at86rf2xx/at86rf2xx_netdev.c [hunk 4]
```diff
@@ -130,59 +148,115 @@ static int _send(netdev_t *netdev, const iolist_t *iolist)
...
-#ifndef MODULE_AT86RF231
-        at86rf2xx_fb_read(dev, &(rssi), 1);
         at86rf2xx_fb_stop(dev);
+        rssi = at86rf2xx_reg_read(dev, AT86RF2XX_REG__PHY_ED_LEVEL);
 #else
+        at86rf2xx_fb_read(dev, &(radio_info->lqi), 1);
+        at86rf2xx_fb_read(dev, &(rssi), 1);
         at86rf2xx_fb_stop(dev);
-        rssi = at86rf2xx_reg_read(dev, AT86RF2XX_REG__PHY_ED_LEVEL);
+        rssi = 3 * rssi;
```
Comment (kYc0o) [inline drivers/at86rf2xx/at86rf2xx_netdev.c]: The changes here and above after `#elif` I think are out of scope, if I'm not wrong, they "fix" the behaviour on at86rf2xx devices, isn't it?

File: drivers/at86rf2xx/at86rf2xx_netdev.c [hunk 5]
```diff
@@ -130,59 +148,115 @@ static int _send(netdev_t *netdev, const iolist_t *iolist)
...
     else {
         at86rf2xx_fb_stop(dev);
     }
-
+#ifdef MODULE_AT86RFR2
+    /* All data is fetched
+     * clear frame buffer protection
+     */
+    *AT86RF2XX_REG__TRX_CTRL_2 &= ~(1 << RX_SAFE_MODE);
+#endif
```
Comment (kYc0o) [inline drivers/at86rf2xx/at86rf2xx_netdev.c]: Again, only for at86rfr2?

File: drivers/at86rf2xx/at86rf2xx_netdev.c [hunk 6]
```diff
@@ -618,11 +702,18 @@ static void _isr(netdev_t *netdev)
                  || (state == AT86RF2XX_STATE_BUSY_TX_ARET)) {
             /* check for more pending TX calls and return to idle state if
              * there are none */
-            assert(dev->pending_tx != 0);
+#ifdef MODULE_AT86RFR2
+            /* clear TX interrupt status */
+            dev->irq_status &= ~AT86RF2XX_IRQ_STATUS_MASK__TX_END;
+#endif
+            // assert(dev->pending_tx != 0);
```
Comment (kYc0o) [inline drivers/at86rf2xx/at86rf2xx_netdev.c]: Please remove C++ comments.

File: drivers/at86rf2xx/at86rfr2_interrupts.c [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+#endif
+
+#define DEBUG_ATRFR2_PINS (0)
+#ifdef DEBUG_ATRFR2_PINS
+    #define DEBUG_ATRFR2_PORT_DDR     (DDRF)
+    #define DEBUG_ATRFR2_PORT         (PORTF)
+    #define DEBUG_ATRFR2_PIN_TX_START (1 << PORTF7)
+    #define DEBUG_ATRFR2_PIN_TX_END   (1 << PORTF6)
+    #define DEBUG_ATRFR2_PIN_RX_END   (1 << PORTF5)
+#endif
```
Comment (kYc0o) [inline drivers/at86rf2xx/at86rfr2_interrupts.c]: You might remove the indentation of your `#define`s here and above.

File: drivers/at86rf2xx/at86rfr2_interrupts.c [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+#if DEBUG_ATRFR2_PINS
+    /*initialize Debug Pins */
+    /* Port Pin as Output */
+    DEBUG_ATRFR2_PORT_DDR |=  (DEBUG_ATRFR2_PIN_TX_START
+                               | DEBUG_ATRFR2_PIN_TX_END | DEBUG_ATRFR2_PIN_RX_END);
+    /* Pin Low */
+    DEBUG_ATRFR2_PORT     &= ~(DEBUG_ATRFR2_PIN_TX_START
+                               | DEBUG_ATRFR2_PIN_TX_END | DEBUG_ATRFR2_PIN_RX_END);
+#endif
+}
```
Comment (kYc0o) [inline drivers/at86rf2xx/at86rfr2_interrupts.c]: If all the flags are not set (`RXTX_LED_ENABLE` and `DEBUG_ATRFR2_PINS`) this function becomes empty? Shouldn't the check for those variables be done by the caller?

File: drivers/at86rf2xx/at86rfr2_interrupts.c [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+{
+    if (sched_context_switch_request) {
+        thread_yield();
+        thread_yield_isr();
+    }
+    __exit_isr();
+}
+
+/**
+ * \brief ISR for transceiver's receive end interrupt
```
Comment (kYc0o) [inline drivers/at86rf2xx/at86rfr2_interrupts.c]: For consistency in documentation, we prefer the use of `@brief`, here and below.

File: drivers/at86rf2xx/include/at86rf2xx_internal.h [hunk 10]
```diff
@@ -90,7 +108,12 @@ uint8_t at86rf2xx_reg_read(const at86rf2xx_t *dev, uint8_t addr);
  * @param[in] addr      address of the register to write
  * @param[in] value     value to write to the given register
  */
+#ifdef MODULE_AT86RFR2
+void at86rf2xx_reg_write(const at86rf2xx_t *dev, volatile uint8_t *addr,
+                         const uint8_t value);
+#else
```
Comment (kYc0o) [inline drivers/at86rf2xx/include/at86rf2xx_internal.h]: I'd recommend to change the name of the functions to `at86rfr2_reg_*` for the sake of consistency, since the API is changed.

File: drivers/at86rf2xx/include/at86rf2xx_registers.h [hunk 11]
```diff
@@ -50,11 +53,151 @@ extern "C" {
...
 #endif
 /** @} */
 
+/*
+ * For external transceiver SPI and all Register bits have to be defined
+ * For SOC systems we use the definition of the iomXXrfrX
+ */
+#ifdef MODULE_AT86RFR2 /* Begin SOC transceiver */
+
+    #include <avr/io.h>
```
Comment (kYc0o) [inline drivers/at86rf2xx/include/at86rf2xx_registers.h]: Please remove indentation.

File: drivers/at86rf2xx/include/at86rf2xx_registers.h [hunk 12]
```diff
@@ -50,11 +53,151 @@ extern "C" {
...
+
+/**
+ * @name   Bitfield definitions for the IRQ_MASK1/IRQ_STATUS1 register
+ * @{
+ */
+    #define AT86RF2XX_IRQ_STATUS_MASK1__TX_START                 (0x01)
+    #define AT86RF2XX_IRQ_STATUS_MASK1__MAF_0_AMI                (0x02)
+    #define AT86RF2XX_IRQ_STATUS_MASK1__MAF_1_AMI                (0x04)
+    #define AT86RF2XX_IRQ_STATUS_MASK1__MAF_2_AMI                (0x08)
+    #define AT86RF2XX_IRQ_STATUS_MASK1__MAF_3_AMI                (0x10)
```
Comment (kYc0o) [inline drivers/at86rf2xx/include/at86rf2xx_registers.h]: Please remove indentation and C++ comments here and above.

File: drivers/at86rf2xx/include/at86rf2xx_registers.h [hunk 13]
```diff
@@ -361,6 +514,18 @@ extern "C" {
 #define AT86RF2XX_CSMA_SEED_1__CSMA_SEED_1                      (0x07)
 /** @} */
 
+/**
+ * @name   Bitfield definitions for the  TRXPR  Transceiver Pin Register
+ * @{
+ */
+#ifdef MODULE_AT86RFR2
```
Comment (kYc0o) [inline drivers/at86rf2xx/include/at86rf2xx_registers.h]: I'd say wrap also the doxygen in the `#ifdef`.

File: sys/auto_init/netif/auto_init_at86rf2xx.c [hunk 14]
```diff
@@ -37,12 +39,23 @@
  * @brief   Define stack parameters for the MAC layer thread
  * @{
  */
-#define AT86RF2XX_MAC_STACKSIZE     (THREAD_STACKSIZE_DEFAULT)
+#ifndef AT86RF2XX_MAC_STACKSIZE
+    #define AT86RF2XX_MAC_STACKSIZE     (THREAD_STACKSIZE_DEFAULT)
```
Comment (kYc0o) [inline sys/auto_init/netif/auto_init_at86rf2xx.c]: Please remove indentation.

File: sys/auto_init/netif/auto_init_at86rf2xx.c [hunk 15]
```diff
@@ -37,12 +39,23 @@
...
 #ifndef AT86RF2XX_MAC_PRIO
 #define AT86RF2XX_MAC_PRIO          (GNRC_NETIF_PRIO)
 #endif
 
+#ifdef MODULE_AT86RFR2
+/* The atmegarfr2 has a at86rf2xx included with register access no spi required.
+ * As of now it is not intended to support external at86rf2xx devices at the
+ * same time as the On-Chip transceiver drive is in use.
+ */
+    #define AT86RF2XX_NUM 1
```
Comment (kYc0o) [inline sys/auto_init/netif/auto_init_at86rf2xx.c]: Please remove indentation.

File: drivers/at86rf2xx/at86rf2xx.c [hunk 16]
```diff
@@ -113,11 +182,13 @@ void at86rf2xx_reset(at86rf2xx_t *dev)
     tmp |= (AT86RF2XX_TRX_CTRL_0_CLKM_CTRL__OFF);
     at86rf2xx_reg_write(dev, AT86RF2XX_REG__TRX_CTRL_0, tmp);
 
+    /* clear interrupt flags, AT86RF233 Manual p.33*/
+    at86rf2xx_reg_read(dev, AT86RF2XX_REG__IRQ_STATUS);
+
```
Comment (kYc0o) [inline drivers/at86rf2xx/at86rf2xx.c]: Unrelated.

File: drivers/at86rf2xx/at86rf2xx_getset.c [hunk 17]
```diff
@@ -332,7 +363,7 @@ void at86rf2xx_set_option(at86rf2xx_t *dev, uint16_t option, bool state)
 
     /* set option field */
     dev->netdev.flags = (state) ? (dev->netdev.flags |  option)
-                                : (dev->netdev.flags & ~option);
+                        : (dev->netdev.flags & ~option);
```
Comment (kYc0o) [inline drivers/at86rf2xx/at86rf2xx_getset.c]: Please revert here and below for the ternary statements.

File: drivers/at86rf2xx/at86rf2xx_internal.c [hunk 18]
```diff
@@ -55,50 +87,80 @@ uint8_t at86rf2xx_reg_read(const at86rf2xx_t *dev, uint8_t addr)
...
     getbus(dev);
     spi_transfer_byte(SPIDEV, CSPIN, true, reg);
     spi_transfer_byte(SPIDEV, CSPIN, true, offset);
     spi_transfer_bytes(SPIDEV, CSPIN, false, NULL, data, len);
     spi_release(SPIDEV);
+#endif
 }
 
-void at86rf2xx_sram_write(const at86rf2xx_t *dev, uint8_t offset,
+void at86rf2xx_sram_write(__attribute__((unused)) const at86rf2xx_t *dev, uint8_t offset,
```
Comment (kYc0o) [inline drivers/at86rf2xx/at86rf2xx_internal.c]: Use `(void)dev;` instead.

File: drivers/at86rf2xx/at86rf2xx_internal.c [hunk 19]
```diff
@@ -55,50 +87,80 @@ uint8_t at86rf2xx_reg_read(const at86rf2xx_t *dev, uint8_t addr)
...
     getbus(dev);
     spi_transfer_byte(SPIDEV, CSPIN, true, reg);
     spi_transfer_byte(SPIDEV, CSPIN, true, offset);
     spi_transfer_bytes(SPIDEV, CSPIN, false, data, NULL, len);
     spi_release(SPIDEV);
+#endif
 }
 
-void at86rf2xx_fb_start(const at86rf2xx_t *dev)
+void at86rf2xx_fb_start(__attribute__((unused)) const at86rf2xx_t *dev)
```
Comment (kYc0o) [inline drivers/at86rf2xx/at86rf2xx_internal.c]: Use `(void)dev;` instead.

File: drivers/at86rf2xx/at86rf2xx_internal.c [hunk 20]
```diff
@@ -55,50 +87,80 @@ uint8_t at86rf2xx_reg_read(const at86rf2xx_t *dev, uint8_t addr)
...
+#endif
+
+#if ENABLE_HEX_DUMP_RX
+    puts("RECEIVED:");
+    od_hex_dump(data, len, OD_WIDTH_DEFAULT);
+#endif
 }
 
-void at86rf2xx_fb_stop(const at86rf2xx_t *dev)
+void at86rf2xx_fb_stop(__attribute__((unused)) const at86rf2xx_t *dev)
```
Comment (kYc0o) [inline drivers/at86rf2xx/at86rf2xx_internal.c]: Use `(void)dev;` instead.

File: drivers/at86rf2xx/at86rf2xx.c [hunk 21]
```diff
@@ -36,6 +37,12 @@
 #define ENABLE_DEBUG (0)
 #include "debug.h"
 
+#include "board.h"
+#define ENABLE_HEX_DUMP_TX  (0)
+#if ENABLE_HEX_DUMP_TX
+#include "od.h"
```
[the remark this answers is not in the document]
Comment (Josar) [reply drivers/at86rf2xx/at86rf2xx.c]: To enable a hexdump at the lowest possible layer.

File: drivers/at86rf2xx/at86rf2xx.c [hunk 22]
```diff
@@ -48,12 +55,34 @@ void at86rf2xx_setup(at86rf2xx_t *dev, const at86rf2xx_params_t *params)
     /* radio state is P_ON when first powered-on */
     dev->state = AT86RF2XX_STATE_P_ON;
     dev->pending_tx = 0;
+
+#ifdef MODULE_AT86RFR2
+    /* Store device pointer for interrupts */
+    at86rfr2_dev = (netdev_t *)dev;
```
[the remark this answers is not in the document]
Comment (Josar) [reply drivers/at86rf2xx/at86rf2xx.c]: `drivers/at86rf2xx/include/at86rf2xx_internal.h`

It is used in the interrupts.

File: drivers/at86rf2xx/at86rf2xx.c [hunk 23]
```diff
@@ -164,6 +235,14 @@ void at86rf2xx_tx_exec(const at86rf2xx_t *dev)
 
     /* write frame length field in FIFO */
     at86rf2xx_sram_write(dev, 0, &(dev->tx_frame_len), 1);
+#if ENABLE_HEX_DUMP_TX
+    uint8_t len = dev->tx_frame_len;
+    uint8_t data[len];
+    memcpy( data, (void *)(AT86RF2XX_REG__TRXFBST), len);
+    puts("SENDING:");
+    od_hex_dump(data, len, OD_WIDTH_DEFAULT);
+#endif
```
[the remark this answers is not in the document]
Comment (Josar) [reply drivers/at86rf2xx/at86rf2xx.c]: When i remember right pkt_dump is used for received packets not for transmitting.

File: drivers/at86rf2xx/at86rf2xx_getset.c [hunk 24]
```diff
@@ -413,13 +454,16 @@ static inline void _set_state(at86rf2xx_t *dev, uint8_t state, uint8_t cmd)
...
+            DEBUG("_set_state: state is: 0x%02x\n", state);
+        }
     }
     /* Although RX_AACK_ON state doesn't get read back,
      * at least make sure if state transition is in progress or not
      */
     else {
-        while (at86rf2xx_get_status(dev) == AT86RF2XX_STATE_IN_PROGRESS) {}
+        while (at86rf2xx_get_status(dev) == AT86RF2XX_STATE_IN_PROGRESS) {
+        }
```
[the remark this answers is not in the document]
Comment (Josar) [reply drivers/at86rf2xx/at86rf2xx_getset.c]: That was the old uncrustify file! This slipped my eyes.

File: drivers/at86rf2xx/at86rf2xx.c (hunk 23 above)
[the remark this answers is not in the document]
Comment (miri64) [reply drivers/at86rf2xx/at86rf2xx.c]: Nope, its used for all kinds of packets ;-).

[the remark this answers is not in the document]
Comment (miri64) [reply drivers/at86rf2xx/at86rf2xx.c]: Ah, but not for `netdev` data! Only `gnrc` internal packets.

File: drivers/at86rf2xx/at86rf2xx_netdev.c (hunk 2 above)
Comment (Josar) [reply drivers/at86rf2xx/at86rf2xx_netdev.c]: Normally `at86rf2xx_fb_stop(dev);` would release the frame buffer, but in this implementation i do not release the frame buffer in `at86rf2xx_fb_stop(dev);`, so it has to be done additionally.
And it is in a `#ifdef MODULE_AT86RFR2`.

File: drivers/at86rf2xx/at86rf2xx_netdev.c (hunk 1 above)
Comment (Josar) [reply drivers/at86rf2xx/at86rf2xx_netdev.c]: To clarify the progblem. this function gets called 2 times, one time to get back the size for the buffer and one time to fetch the data. But the frame bufer protection is allready released after the size is determined. Which could result in data loss when a node send data meanwhile.

File: drivers/at86rf2xx/at86rfr2_interrupts.c (hunk 8 above)
Comment (Josar) [reply drivers/at86rf2xx/at86rfr2_interrupts.c]: Then i have to check it twice, no big deal. but i thin  we can relly on the compiler to remove the empty functioncall.

File: drivers/at86rf2xx/include/at86rf2xx_registers.h (hunk 13 above)
Comment (Josar) [reply drivers/at86rf2xx/include/at86rf2xx_registers.h]: Just for consistency, this is done like 
https://github.com/Josar/RIOT/blob/32b72b4de248b50d314e40b650b8a3f874baa955/drivers/at86rf2xx/include/at86rf2xx_registers.h#L499

File: drivers/at86rf2xx/at86rf2xx.c (hunk 23 above)
[the remark this answers is not in the document]
Comment (Josar) [reply drivers/at86rf2xx/at86rf2xx.c]: It´s been a while, but i remember that i had been missing some info, this is why i added this dumps.

[the remark this answers is not in the document]
Comment (kYc0o) [reply drivers/at86rf2xx/at86rf2xx.c]: I'd say a reuse of pkt_dump would make sense. On the other hand, I might find useful dumping packets at this level, but only at the driver development phase, afterwards for me pkt_dump would make more sense. However, I don't know how useful information below the gnrc layer is needed/important to display while transmitting packets. It's strange that we didn't need this in the past, especially for the at86rf2xx driver.

File: drivers/at86rf2xx/at86rf2xx_getset.c (hunk 24 above)
[the remark this answers is not in the document]
Comment (kYc0o) [reply drivers/at86rf2xx/at86rf2xx_getset.c]: No problem :-)

File: drivers/at86rf2xx/at86rf2xx_netdev.c (hunk 4 above)
Comment (kYc0o) [reply drivers/at86rf2xx/at86rf2xx_netdev.c]: Yeah, that's nice, but it should come in a follow up PR, since this one is about adding support for the at86rfr2 driver.

File: drivers/at86rf2xx/include/at86rf2xx_internal.h (hunk 10 above)
Comment (Josar) [reply drivers/at86rf2xx/include/at86rf2xx_internal.h]: This has to be done when rebasing i think. But good to be alerted.

File: drivers/at86rf2xx/at86rf2xx_netdev.c (hunk 1 above)
Comment (gebart) [reply drivers/at86rf2xx/at86rf2xx_netdev.c]: The expected behavior is well defined in http://api.riot-os.org/structnetdev__driver.html#ae2c8cad80067e3b1f9979931ddb3cc8b
I agree there is a problem with the current implementation, this should be fixed for all variants, but in a separate PR. I think the proper fix is to only call at86rf2xx_fb_stop() when len > 0 (see docs for netdev_driver_t recv

File: drivers/at86rf2xx/at86rf2xx.c (hunk 23 above)
[the remark this answers is not in the document]
Comment (Josar) [reply drivers/at86rf2xx/at86rf2xx.c]: Lets say, i did not write the upper layers and will ensure that what is packed is well formatted.
Then i have to get the raw data put to the transmitter. Only then can i check that the data put to the transceiver is as i would expect it.
And it is a debugging message which is seperatly enable, it only afffects the build if someone needs to check that what happend in the upper layers results in the expected packets.

File: drivers/at86rf2xx/at86rf2xx_netdev.c (hunk 1 above)
Comment (Josar) [reply drivers/at86rf2xx/at86rf2xx_netdev.c]: I will prepare this, can we merge that beforehand, so that a can adjust the at86rfr2 driver rebased on that?

File: drivers/at86rf2xx/at86rf2xx_netdev.c (hunk 4 above)
Comment (Josar) [reply drivers/at86rf2xx/at86rf2xx_netdev.c]: I will prepare this, can we merge that beforehand, so that a can adjust the at86rfr2 driver rebased on that?

**Comment by Josar** [comment]
https://github.com/RIOT-OS/RIOT/pull/9508

File: drivers/at86rf2xx/at86rf2xx.c [hunk 25]
```diff
@@ -85,22 +114,62 @@ void at86rf2xx_reset(at86rf2xx_t *dev)
     at86rf2xx_set_option(dev, AT86RF2XX_OPT_CSMA, true);
     at86rf2xx_set_option(dev, AT86RF2XX_OPT_TELL_RX_START, false);
     at86rf2xx_set_option(dev, AT86RF2XX_OPT_TELL_RX_END, true);
+
+    /* enable safe mode (protect RX FIFO until reading data starts) */
+    at86rf2xx_reg_write(dev, AT86RF2XX_REG__TRX_CTRL_2,
+                        AT86RF2XX_TRX_CTRL_2_MASK__RX_SAFE_MODE);
```
[the remark this answers is not in the document]
Comment (Josar) [reply drivers/at86rf2xx/at86rf2xx.c]: This was just replaced, as it was buried in a lot of ifdefs.

File: drivers/at86rf2xx/at86rf2xx_netdev.c (hunk 4 above)
Comment (kYc0o) [reply drivers/at86rf2xx/at86rf2xx_netdev.c]: Yes, if you want to clean up at86rf2xx code first it would be much easier to review this PR.

**Comment by kYc0o** [comment]
@Josar you may rebase this PR for a final review.

**Comment by Josar** [comment]
@kYc0o squashed and rebased.

**Comment by benpicco** [comment]
I now have a board with this radio #12411
It's [much cheaper](https://www.ebay.de/itm/282736110364) than the official dev boards too.

**Comment by benpicco** [comment]
Superseded by #12537
